pkg/adapter/nimbus-k8tls/manager/cronjob.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Authors of Nimbus
+
+package manager
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	intentv1 "github.com/5GSEC/nimbus/api/v1"
+	adapterutil "github.com/5GSEC/nimbus/pkg/adapter/util"
+)
+
+func createOrUpdateCronJob(ctx context.Context, logger logr.Logger, cwnp intentv1.ClusterNimbusPolicy, cronJob *batchv1.CronJob) {
+	cronJob.Spec.JobTemplate.Spec.Template.Spec.ServiceAccountName = NamespaceName
+	if err := ctrl.SetControllerReference(&cwnp, cronJob, scheme); err != nil {
+		logger.Error(err, "failed to set OwnerReference on Kubernetes CronJob", "CronJob.Name", cronJob.Name)
+		return
+	}
+
+	var existingCronJob batchv1.CronJob
+	err := k8sClient.Get(ctx, types.NamespacedName{Name: cronJob.Name, Namespace: cronJob.Namespace}, &existingCronJob)
+	if err != nil && !errors.IsNotFound(err) {
+		logger.Error(err, "failed to get Kubernetes CronJob", "CronJob.Name", cronJob.Name, "CronJob.Namespace", cronJob.Namespace)
+		return
+	}
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if err = k8sClient.Create(ctx, cronJob); err != nil {
+				logger.Error(err, "failed to create Kubernetes CronJob", "CronJob.Name", cronJob.Name, "CronJob.Namespace", cronJob.Namespace)
+				return
+			}
+			logger.Info("created Kubernetes CronJob", "CronJob.Name", cronJob.Name, "CronJob.Namespace", cronJob.Namespace)
+		}
+	} else {
+		cronJob.ResourceVersion = existingCronJob.ResourceVersion
+		cronJob.Labels = existingCronJob.Labels
```

Comment:
What about the case where the labels on the cwnp changed ? In that case, is it correct to overwrite the labels with those of the existing CronJob ?

Reply:
In any case, is it really needed to have a label on a cronJob ? Not sure what it does..

Reply:
Removed.